src/Session/Onboarding/OnboardingSessionManager.php

```diff
@@ -0,0 +1,275 @@
+            throw $exception;
+        }
+
+        // Exceptions only for transit actions
+        if ($action === 'transit') {
+            if (!$this->getOpened()) {
+                $exception = new PsCheckoutSessionException($genericErrorMsg . 'Unable to find an opened session', PsCheckoutSessionException::OPENED_SESSION_NOT_FOUND);
+                $module->getLogger()->error('Unable to find an opened session', ['exception' => $exception, 'trace' => $exception->getTraceAsString()]);
+                throw $exception;
+            }
+
+            $authorizedTransition = false;
+
+            if (is_array($nextTransition['from'])) {
+                foreach ($nextTransition['from'] as $transition) {
+                    if ($this->getOpened()->getStatus() === $transition) {
+                        $authorizedTransition = true;
+                        break;
+                    }
+                }
+            }
+
+            if ($this->getOpened()->getStatus() === $nextTransition['from']) {
+                $authorizedTransition = true;
+            }
+
+            if (!$authorizedTransition) {
+                $exception = new PsCheckoutSessionException($genericErrorMsg . 'The session is not authorized to transit from ' . $this->getOpened()->getStatus() . ' to ' . $nextTransition['to'], PsCheckoutSessionException::FORBIDDEN_SESSION_TRANSITION);
+                $module->getLogger()->error('The session transition is not authorized', ['from' => $this->getOpened()->getStatus(), 'to' => $nextTransition['to'], 'exception' => $exception, 'trace' => $exception->getTraceAsString()]);
+                throw $exception;
+            }
+        }
+
+        if ($updateIntersect !== $sortedUpdateConfiguration) {
+            $exception = new PsCheckoutSessionException($genericErrorMsg . 'Missing expected update session parameters.' . PHP_EOL . 'Transition : ' . $next, PsCheckoutSessionException::MISSING_EXPECTED_PARAMETERS);
+            $module->getLogger()->error($exception->getMessage(), ['transition' => $nextTransition, 'update' => $update, 'updateIntersect' => $updateIntersect, 'sortedUpdateConfiguration' => $sortedUpdateConfiguration, 'exception' => $exception, 'trace' => $exception->getTraceAsString()]);
+            throw $exception;
+        }
+    }
+
+    /**
+     * Apply the onboarding session transition
+     *
+     * @param string $next Next state to transit
+     * @param array $update Session data to update
+     *
+     * @return \PrestaShop\Module\PrestashopCheckout\Session\Session
+     *
+     * @throws \PrestaShop\Module\PrestashopCheckout\Exception\PsCheckoutSessionException
+     */
+    public function apply($next, array $update)
+    {
+        $this->can($next, $update);
+
+        $nextTransition = $this->transitions[$next];
+        $session = $this->getOpened();
+
+        foreach ($update as $updateKey => $updateValue) {
```

Comment:
Need to check if the function works properly. For example when logging in with PayPal the state changes to "onboard_paypal" and the session "expires_at" changes to false so it's empty in the database. 
![image](https://user-images.githubusercontent.com/25088641/153574310-8872160d-f7c0-4bdd-9a41-83d6be5844b4.png)
in this case $updateConfigValue is false and the **if** checks if it's not null. Since the false !== null the expires_at value is set to false instead of the $updateValue that should be correct.